ouroboros-consensus/src/Ouroboros/Consensus/Protocol/Abstract.hs

```diff
@@ -134,10 +134,10 @@ class ( Show (ChainState    p)
   --
   -- Note: we do not assume that the candidate fragments fork less than @k@
   -- blocks back.
-  preferCandidate :: HasHeader b
+  preferCandidate :: HasHeader hdr
```

Comment:
Should we use `SupportedHeader` here too?
Could/should we include `HasHeader` as a mandatory constraint for headers?

Reply:
We could, those we don't currently need it. But I guessthere is no reason not to add it.